ynr/apps/people/forms/forms.py

```diff
@@ -206,6 +211,11 @@ class Meta:
     )
 
     def save(self, commit=True):
+        self.instance.remove_membership = self.cleaned_data["remove_membership"]
+        if self.instance.remove_membership:
+            self.instance.delete()
+            return None
+
```

Comment:
So after doing [some research](https://docs.djangoproject.com/en/4.1/topics/forms/formsets/#can-delete) I found that this was already being managed by Django. 

We need to make a couple of changes to this commit:

1. In person form 
```diff 
-                  {{ membership_form.remove_membership }}
-                  <span>{{ membership_form.remove_membership.label }}</span>
+                  {{ membership_form.DELETE }}
+                  <span>Delete this candidacy</span>
```
(The `DELETE` field is added because we have `can_delete=True` on the FormSet)

2. Remove the field you added:
```diff
-        self.fields["remove_membership"] = forms.BooleanField(
-            label="Delete this candidacy", initial=False, required=False
-        )
```
3. Remove the `save()` logic:
```diff
-        self.instance.remove_membership = self.cleaned_data["remove_membership"]
-        if self.instance.remove_membership:
-            self.instance.delete()
-            return None
-
```

So, nicely this results in _less_ code to do more or less the same thing!

Reply:
Amazing find, thank you! Great lesson for me in formsets :D This has been addressed in 8aa61461